### PR TITLE
delete ineffectual assginment

### PR DESCRIPTION
**Pull request by zouyee**
**What this PR does / why we need it**:
delete ineffectual assignment
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/kubelet/certificate/bootstrap/bootstrap.go [hunk 1]
```diff
@@ -60,6 +62,14 @@ func LoadClientCert(kubeconfigPath string, bootstrapPath string, certDir string,
 	if err != nil {
 		return fmt.Errorf("unable to load bootstrap kubeconfig: %v", err)
 	}
+	client, err := kubernetes.NewForConfig(bootstrapClientConfig)
+	if err != nil {
+		return err
+	}
+	_, err = client.ServerVersion()
+	if err != nil {
```
Comment (supereagle) [inline pkg/kubelet/certificate/bootstrap/bootstrap.go]: `if _, err := client.ServerVersion(); err != nil` is much better?

File: pkg/kubelet/certificate/bootstrap/bootstrap.go [hunk 2]
```diff
@@ -86,9 +96,16 @@ func LoadClientCert(kubeconfigPath string, bootstrapPath string, certDir string,
 	if err != nil {
 		return err
 	}
-	if err := certutil.WriteCert(certPath, certData); err != nil {
+	if err = certutil.WriteCert(certPath, certData); err != nil {
```
Comment (supereagle) [inline pkg/kubelet/certificate/bootstrap/bootstrap.go]: Maybe this change is not needed.

File: pkg/kubelet/certificate/bootstrap/bootstrap.go [hunk 3]
```diff
@@ -86,9 +96,16 @@ func LoadClientCert(kubeconfigPath string, bootstrapPath string, certDir string,
...
 		return err
 	}
+	cert, err := x509.ParseCertificate(certData)
+	if err != nil {
+		return err
+	}
+	if cert.NotAfter.Sub(time.Now()) <= time.Duration(0) {
```
Comment (supereagle) [inline pkg/kubelet/certificate/bootstrap/bootstrap.go]: This has logic error? When `cert.NotAfter.Sub(time.Now()) > time.Duration(0)`, the cert will be expired.

At the same time, `time` package provides APIs to compare time, can directly use `cert.NotAfter.After(time.Now())`.

File: pkg/kubelet/certificate/bootstrap/bootstrap.go [hunk 4]
```diff
@@ -62,7 +62,11 @@ func LoadClientCert(kubeconfigPath string, bootstrapPath string, certDir string,
 	if err != nil {
 		return fmt.Errorf("unable to load bootstrap kubeconfig: %v", err)
 	}
-	_, err = kubernetes.NewForConfig(bootstrapClientConfig)
+	client, err := kubernetes.NewForConfig(bootstrapClientConfig)
+	if err != nil {
+		return err
+	}
+	_, err = client.ServerVersion()
```
Comment (liggitt) [inline pkg/kubelet/certificate/bootstrap/bootstrap.go]: as commented in https://github.com/kubernetes/kubernetes/pull/50298, I don't think these extra checks are necessary, we'll already return errors when we try to use the API if we can't reach it

File: pkg/kubelet/certificate/bootstrap/bootstrap.go [hunk 5]
```diff
@@ -86,9 +92,16 @@ func LoadClientCert(kubeconfigPath string, bootstrapPath string, certDir string,
...
 	}
-	if err := certutil.WriteCert(certPath, certData); err != nil {
+	if err = certutil.WriteCert(certPath, certData); err != nil {
+		return err
+	}
+	cert, err := x509.ParseCertificate(certData)
+	if err != nil {
 		return err
 	}
+	if cert.NotAfter.Sub(time.Now()) <= time.Duration(0) {
```
Comment (liggitt) [inline pkg/kubelet/certificate/bootstrap/bootstrap.go]: as commented in https://github.com/kubernetes/kubernetes/pull/50298, client side validation of cert expiry and refusal to create the client isn't how we want to handle this. there is client cert rotation that can request a new cert in this condition, but that is not done here

**Pull request merged**